### DISCUSSION TITLE
Is main-thread intended to be the same for all instances of a plugin?

### DISCUSSION THREAD

**Discussion by alien-brother** [free-audio/clap#435] [Q&A] [answered]
`thread-check.h` seems to imply that on Windows and MacOS, yes, since it is the UI thread. However, what is intended on Linux?

This has impact if plugin instances share resources. For example, could be global state of a UI framework or another 3rd party library.

**Comment by baconpaul** [comment, marked as answer]
The documentation talks about main thread as “the” thread and there is certainly the assumption that at a given moment only one thread is the main thread. So even though the documentation doesn’t say this, I believe that within a single process boundary on Linux you can assume main thread functions are always executed in a single threaded fashion (even though a host may I suppose choose to migrate its main thread)

if my colleagues agree I propose we expand the thread check documentation to include this fact since the Linux ui is not similarly single threaded architecturally (as I’m sure you know hence the question)
